### PR TITLE
Expand payload unpermitted_parameters.action_controller

### PR DESCRIPTION
**Pull request by bbuchalter**
Previously, when emitting the event, the payload did not include enough information to know which controller action recieved the unpermitted paramaters. This limits the usefulness of the events. This PR expands the payload of the event to include everything provided by the start_processing.action_controller event.

### CODE REVIEW

File: actionpack/lib/action_controller/metal/instrumentation_payload.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module ActionController
+  # Encasulates logic needed to contruct payload for controller events.
+  module InstrumentationPayload
+    private
+      # The payload data used by the following instrumented events:
+      # * start_processing.action_controller
+      # * process_action.action_controller
+      def controller_instrumentation_payload
```
Comment (rafaelfranca) [inline actionpack/lib/action_controller/metal/instrumentation_payload.rb:10]: I don't think this duplication is worth extracting another module. Can you revert this change and let the duplication exist? Also I don't think `unpermitted_parameters.action_controller` event needs the headers or the format for example, so the duplication will not be so big.

File: guides/source/active_support_instrumentation.md [hunk 2]
```diff
@@ -280,9 +280,17 @@ INFO. Additional keys may be added by the caller.
...
-| Key     | Value            |
-| ------- | ---------------- |
-| `:keys` | Unpermitted keys |
+| Key             | Value                                                     |
+| --------------- | --------------------------------------------------------- |
+| `:keys`         | Unpermitted keys                                          |
+| `:controller`   | The controller name                                       |
+| `:action`       | The action                                                |
+| `:params`       | Hash of request parameters without any filtered parameter |
+| `:headers`      | Request headers                                           |
```
Comment (rafaelfranca) [inline guides/source/active_support_instrumentation.md:289]: I'd not include this, and if users want they already have the request.
```suggestion
```

File: guides/source/active_support_instrumentation.md [hunk 3]
```diff
@@ -280,9 +280,17 @@ INFO. Additional keys may be added by the caller.
...
+| `:format`       | html/js/json/xml etc                                      |
```
Comment (rafaelfranca) [inline guides/source/active_support_instrumentation.md:290]: I don't think the `format` is useful to debug unpermitted parameters, so we can remove it.
```suggestion                                   |
```

File: guides/source/active_support_instrumentation.md [hunk 4]
```diff
@@ -280,9 +280,17 @@ INFO. Additional keys may be added by the caller.
...
+| `:method`       | HTTP request verb                                         |
+| `:path`         | Request path                                              |
```
Comment (rafaelfranca) [inline guides/source/active_support_instrumentation.md:292]:
```suggestion
```

File: guides/source/active_support_instrumentation.md [hunk 5]
```diff
@@ -280,9 +280,17 @@ INFO. Additional keys may be added by the caller.
...
-| `:keys` | Unpermitted keys |
+| Key             | Value                                                     |
+| --------------- | --------------------------------------------------------- |
+| `:keys`         | Unpermitted keys                                          |
+| `:controller`   | The controller name                                       |
+| `:action`       | The action                                                |
+| `:params`       | Hash of request parameters without any filtered parameter |
+| `:headers`      | Request headers                                           |
+| `:format`       | html/js/json/xml etc                                      |
+| `:method`       | HTTP request verb                                         |
```
Comment (rafaelfranca) [inline guides/source/active_support_instrumentation.md:291]:
```suggestion
```

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

**Comment by bbuchalter** [comment]
Superseded by https://github.com/rails/rails/pull/41809
